src/Aspire.Hosting/Dashboard/DashboardOptions.cs

```diff
@@ -11,23 +11,69 @@ internal class DashboardOptions
+    public DashboardAuthMode DashboardAuthMode { get; set; }
     public string? DashboardToken { get; set; }
+    public OpenIdConnectPolicyOptions? OpenIdConnect { get; set; } = new();
+    public OpenIdConnectSettings? OpenIdConnectSettings { get; set; } = new();
     public string? OtlpGrpcEndpointUrl { get; set; }
     public string? OtlpHttpEndpointUrl { get; set; }
     public string? OtlpApiKey { get; set; }
     public string AspNetCoreEnvironment { get; set; } = "Production";
 }
 
+internal enum DashboardAuthMode
+{
+    Unsecured,
+    OpenIdConnect,
+    BrowserToken
+}
+
+internal sealed class OpenIdConnectPolicyOptions
+{
+    public string NameClaimType { get; set; } = "name";
+    public string UsernameClaimType { get; set; } = "preferred_username";
+
+    /// <summary>
+    /// Gets the optional name of a claim that users authenticated via OpenID Connect are required to have.
+    /// If specified, users without this claim will be rejected. If <see cref="RequiredClaimValue"/>
+    /// is also specified, then the value of this claim must also match <see cref="RequiredClaimValue"/>.
+    /// </summary>
+    public string RequiredClaimType { get; set; } = "";
+
+    /// <summary>
+    /// Gets the optional value of the <see cref="RequiredClaimType"/> claim for users authenticated via
+    /// OpenID Connect. If specified, users not having this value for the corresponding claim type are
+    /// rejected.
+    /// </summary>
+    public string RequiredClaimValue { get; set; } = "";
+}
+
+internal class OpenIdConnectSettings
+{
+    public string? Authority { get; set; }
+    public string? MetadataAddress { get; set; }
+    public string? ClientId { get; set; }
+    public string? ClientSecret { get; set; }
+    public ICollection<string> Scope { get; } = [];
+}
+
 internal class ConfigureDefaultDashboardOptions(IConfiguration configuration, IOptions<DcpOptions> dcpOptions) : IConfigureOptions<DashboardOptions>
 {
     public void Configure(DashboardOptions options)
     {
         options.DashboardPath = dcpOptions.Value.DashboardPath;
-        options.DashboardUrl = configuration["ASPNETCORE_URLS"];
-        options.DashboardToken = configuration["AppHost:BrowserToken"];
+        options.DashboardUrl = configuration[KnownConfigNames.AspNetCoreUrls];
 
-        options.OtlpGrpcEndpointUrl = configuration["DOTNET_DASHBOARD_OTLP_ENDPOINT_URL"];
-        options.OtlpHttpEndpointUrl = configuration["DOTNET_DASHBOARD_OTLP_HTTP_ENDPOINT_URL"];
+        if (Enum.TryParse<DashboardAuthMode>(configuration[DashboardConfigNames.DashboardFrontendAuthModeName.ConfigKey], out var dashboardAuthMode))
+        {
+            options.DashboardAuthMode = dashboardAuthMode;
+        }
+
+        options.DashboardToken = configuration["AppHost:BrowserToken"];
+        configuration.Bind("Dashboard:Frontend:OpenIdConnect", options.OpenIdConnect);
+        configuration.Bind("Authentication:Schemes:OpenIdConnect", options.OpenIdConnectSettings);
```

Comment:
I agree with @JamesNK comment on the earlier PR that these two related sections seem a bit too disconnected. Just because this is the section normal web apps use to configure OpenIdConnect doesn't mean that we have to do that for the dashboard, right? Could we include everything under "Dashboard:Frontend:OpenIdConnect"? We could forward it to "Authentication:Schemes:OpenIdConnect" in the orchestrated apps.

Another option might be to add the `NameClaimType`/`UsernameClaimType`/`RequiredClaimType`/`RequiredClaimValue` logic from aspire to ASP.NET Core's OIDC config binding logic and include everything under "Authentication:Schemes:OpenIdConnect" except for the `AuthMode=OpenIdConnect` part.

Reply:
Is it really just the configuration keys you think need to be changed over?

Currently the appsettings config would look something like this...

```
"Dashboard": {
  "Frontend": {
    "AuthMode": "OpenIdConnect",
    "OpenIdConnect": {
      "RequiredClaimType": "",
      "RequiredClaimValue": ""
    }
  }
},
"Authentication": {
  "Schemes": {
    "OpenIdConnect": {
      "Scope": [ "openid", "profile" ],
      "MetadataAddress": "",
      "ClientId": "",
      "ClientSecret": ""
    }
  }
}
```

I'm guessing you'd rather see something like....

```
"Dashboard": {
  "AuthMode": "OpenIdConnect",
  "OpenIdConnect": {
    "Policy": {
      "RequiredClaimType": "",
      "RequiredClaimValue": ""
    },
    "Settings": {
      "Scope": [ "openid", "profile" ],
      "MetadataAddress": "",
      "ClientId": "",
      "ClientSecret": ""
    }
  }
}
```

Reply:
@halter73 Let me know if I’m interpreting this correctly and I will make the changes

Reply:
Sorry for the delay. I just got back from vacation. I don't have much background on this PR, I only dropped a review because @eerhardt requested it. I have a deeper understanding of the inner workings of the OpenIdConnectHandler than the Aspire dashboard.

Is the intention of this PR to only enable/configure OIDC for the dashboard? Or can this configuration also be used by the projects orchestrated by aspire?

If it's the former, I would like to see something like your second example. It makes it much more clear the configuration only applies to the dashboard. The "AuthMode" might not even be necessary since the presence of the "Dashboard:OpenIdConnect" configuration section ought to be enough to indicate that you want to use it.

However, if the idea is to be able to share OIDC configuration between the dashboard and other projects, it does make more sense to have it split out the way you currently have it. But then, I don't see what things like `NameClaimType` and `UsernameClaimType` would be specific to the dashboard. Wouldn't you want to share that between all projects?

I can see why `RequiredClaimType` and `RequiredClaimValue` might be specific to the dashboard, but I'm not sure that would belong in an "OpenIdConnect" subsection. What if we added support for another authentication mode later that also provides claims? Would it be more future proof to put `RequiredClaimType` and `RequiredClaimValue` in a "Dashboard:Frontend:AuthorizationPolicy" section to avoid duplication?

Reply:
This is just for the dashboard, that’s why I thought it made sense to put it in the normal location you’d use for any app instead of encapsulating it.  If there was a way to forward the entire section to the context then I think we can avoid taking a dependency on the OpenIdConnect package and it should support every option in whatever version of the package the user is consuming.  I tried to sort that out but came up empty.  It seemed like it would be there already since you can take config and bind it, but I don’t see a clear path to just copying the config section since the config builder in the context doesn’t even exist at that point

Reply:
> If there was a way to forward the entire section to the context then I think we can avoid taking a dependency on the OpenIdConnect package and it should support every option in whatever version of the package the user is consuming.

It would be nice if there was a one-liner like `Bind` that supported `Dictonary<string, object>` and created nested dictionaries when necessary, but that's not currently supported. Plus, that's not exactly what we need here since it wouldn't create `UNDERSCORE__DELIMITED__SECTION__NAMES` for the keys.

It doesn't seem like it would be too hard to write custom code that calls `IConfguration.GetChildren()` and builds up the dictionary with the right keys though.

Reply:
Yes!  That’s exactly what I was trying originally but the only way I could think of to do it was feed a root like authentication__ and go grab all the env variables that start with that and bring them over directly.  Doesn’t seem like a very good solution, but it might be doable